code/src/components/ProductList.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import { useSelector } from 'react-redux'
+import { Card } from '../lib/Card'
+import { Error } from './Error'
+import styled from 'styled-components/macro'
+
+export const ProductList = () => {
+  const code = useSelector((state) => state.products.product)
+
+  if (!code) return null
+
+  return (
+    <>
+      {/* 1. This code below is not exactly what we wanted to print out but it works with all products. What we wanted to achieve is part 2 */}
+      {code.product && code.status === 1 && (
+        <Card
+          coverImage={code.product.image_url}
+          secondaryText={code.product.brands}>
+          <Text>
+            <Link a href={`https://world.openfoodfacts.org/product/${code.product.code}/`} target="_blank">More Details</Link>
+            <p>Origin: {code.product.origins}</p>
+            <p>Ingredients: {code.product.ingredients_text}</p>
+          </Text>
+          <FinalText>Thank you and have a healthy life!</FinalText>
+        </Card>
+      )}
+
+      {/*2. The code below works as long as the JSON file contains an ingredients object containing itself "vegan" or "vegetarian". BUT some products don't even have this information, so "vegan" and "vegetarian" are returned as undefined. */}
+      {/* {code.product && code.status === 1 && (
+        <Card
+          coverImage={code.product.image_url}
+          title={code.product.ingredients_text}
+          secondaryText={code.product.brands}>
+          <Text>
+            <Link a href={`https://world.openfoodfacts.org/product/${code.product.code}/`} target="_blank">More Details</Link>
+            <p>Origin: {code.product.origins}</p>
+            <p>Vegan: {code.product.ingredients[0].vegan === "yes" ? <i class="em em-leaves" aria-role="presentation" aria-label="LEAF FLUTTERING IN WIND">🍃</i> : <i class="em em-x" aria-role="presentation" aria-label="CROSS MARK">✖️</i>}</p>
```

Comment:
The problem with this approach is the possibilities of null values (as you suspected). If the product has no ingredients defined and the API returns `null` instead of an empty array, when you try to call `code.product.ingredients[0]` you're essentially calling `null[0]` which will throw an error. 

So you need to be more defensive and handle the possibility of null. There's a few ways you could approach this. 

One is to just add more conditions:

```
code.products.ingredients && code.products.ingredients[0].vegan === "yes" ? 'vegan' : 'not vegan'
```

Or you could default to an empty array using `||`:

```
(code.products.ingredients || [])[0].vegan === "yes" ? 'vegan' : 'not vegan'
```

But it gets kinda messy. I think the best approach for this kind of thing is to create a component which returns appropriately. That way you can use longer if statements and give yourself some more room for code:

```
const VeganIndicator = ({ ingredients }) => {
  if (Array.isArray(ingredients)) {
    if (ingredients[0].vegan === "yes") {
      return <i class="em em-leaves" aria-role="presentation" aria-label="LEAF FLUTTERING IN WIND">🍃</i>
    } else {
      return <i class="em em-x" aria-role="presentation" aria-label="CROSS MARK">✖️</i>
    }
  }

  return <p>Product is missing ingredients information</p>
}
```

And then here in your component, mount the component and pass your product ingredients data as a prop:

```
<p>Vegan: <VeganIndicator ingredients={code.product.ingredients} /></p>
```